app/src/main/java/org/groundplatform/android/usecases/location/ShouldEnableLocationLockUseCase.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.groundplatform.android.usecases.location
+
+import javax.inject.Inject
+import org.groundplatform.android.repository.LocationOfInterestRepository
+import org.groundplatform.android.repository.MapStateRepository
+
+/** Use case to determine if location lock should be enabled for a survey. */
+class ShouldEnableLocationLockUseCase
```

Comment:
A "use case" is a user-visible piece of application behavior, not a function whose return value is used internally by the app. If we want to model this as a use case, it would likely be part of the `ActivateSurveyUseCase`. If that's not possible given the current architecture, then it should live in the ViewModel associated with the home screen setup on survey activation.